File: ios/PizzaDelivery/Sources/Models/Order.swift
import Foundation

struct Order: Codable, Identifiable {
    let id: String
    var userId: String
    var storeId: String
    var storeName: String?
    var items: [OrderItem]
    var status: OrderStatus
    var payment: OrderPayment?
    var delivery: OrderDelivery?
    var total: Double
    var createdAt: String
    
    enum CodingKeys: String, CodingKey {
        case id, userId, storeId, storeName, items, status, payment, delivery, total, createdAt
    }
}

struct OrderItem: Codable, Identifiable {
    var id: String { menuItemId }
    var menuItemId: String
    var name: String
    var quantity: Int
    var customizations: [SelectedCustomization]?
    var price: Double
}

enum OrderStatus: String, Codable {
    case placed
    case confirmed
    case preparing
    case ready
    case pickedUp = "picked_up"
    case onTheWay = "on_the_way"
    case delivered
    case cancelled
    
    var displayName: String {
        switch self {
        case .placed: return "Order Placed"
        case .confirmed: return "Confirmed"
        case .preparing: return "Preparing"
        case .ready: return "Ready for Pickup"
        case .pickedUp: return "Picked Up"
        case .onTheWay: return "On the Way"
        case .delivered: return "Delivered"
        case .cancelled: return "Cancelled"
        }
    }
    
    var icon: String {
        switch self {
        case .placed: return "clock.fill"
        case .confirmed: return "checkmark.circle.fill"
        case .preparing: return "flame.fill"
        case .ready: return "bag.fill"
        case .pickedUp: return "car.fill"
        case .onTheWay: return "bicycle"
        case .delivered: return "checkmark.seal.fill"
        case .cancelled: return "xmark.circle.fill"
        }
    }
    
    var progress: Double {
        switch self {
        case .placed: return 0.15
        case .confirmed: return 0.3
        case .preparing: return 0.45
        case .ready: return 0.6
        case .pickedUp: return 0.75
        case .onTheWay: return 0.85
        case .delivered: return 1.0
        case .cancelled: return 0.0
        }
    }
}

struct OrderPayment: Codable {
    var method: String?
    var status: String?
    var transactionId: String?
    var amount: Double?
}

struct OrderDelivery: Codable {
    var driverId: String?
    var driverName: String?
    var status: String?
    var eta: Int?
    var currentLocation: Location?
}

struct CreateOrderRequest: Codable {
    let storeId: String
    let items: [OrderItemRequest]
    let deliveryAddress: Address
    let paymentMethod: String
}

struct OrderItemRequest: Codable {
    let menuItemId: String
    let quantity: Int
    let customizations: [SelectedCustomization]?
}

File: android/app/src/main/java/com/pizzadelivery/android/data/model/Order.kt
package com.pizzadelivery.android.data.model

import com.google.gson.annotations.SerializedName

data class Order(
    val id: String = "",
    val userId: String = "",
    val storeId: String = "",
    val storeName: String = "",
    val items: List<OrderItem> = emptyList(),
    val status: OrderStatus = OrderStatus.PLACED,
    val payment: PaymentInfo? = null,
    val delivery: DeliveryInfo? = null,
    val subtotal: Double = 0.0,
    val tax: Double = 0.0,
    val deliveryFee: Double = 0.0,
    val discount: Double = 0.0,
    val total: Double = 0.0,
    val createdAt: String = "",
    val updatedAt: String = ""
)

data class OrderItem(
    val menuItemId: String = "",
    val name: String = "",
    val quantity: Int = 1,
    val price: Double = 0.0,
    val customizations: List<SelectedCustomization> = emptyList()
)

enum class OrderStatus {
    @SerializedName("placed") PLACED,
    @SerializedName("confirmed") CONFIRMED,
    @SerializedName("preparing") PREPARING,
    @SerializedName("ready") READY,
    @SerializedName("picked_up") PICKED_UP,
    @SerializedName("on_the_way") ON_THE_WAY,
    @SerializedName("delivered") DELIVERED,
    @SerializedName("cancelled") CANCELLED
}

data class PaymentInfo(
    val id: String = "",
    val method: String = "",
    val status: String = "",
    val transactionId: String = ""
)

data class DeliveryInfo(
    val driverId: String? = null,
    val driverName: String? = null,
    val driverPhone: String? = null,
    val status: DeliveryStatus = DeliveryStatus.ASSIGNED,
    val currentLocation: Location? = null,
    val eta: Int? = null,
    val route: List<Location> = emptyList()
)

enum class DeliveryStatus {
    @SerializedName("assigned") ASSIGNED,
    @SerializedName("picked_up") PICKED_UP,
    @SerializedName("on_the_way") ON_THE_WAY,
    @SerializedName("delivered") DELIVERED
}

data class CreateOrderRequest(
    val storeId: String,
    val items: List<OrderItem>,
    val deliveryAddressId: String,
    val paymentMethodId: String,
    val couponCode: String? = null
)
